avm/res/event-grid/system-topic/main.bicep

```diff
@@ -24,6 +24,24 @@ import { roleAssignmentType } from 'br/public:avm/utl/types/avm-common-types:0.5
 @description('Optional. Array of role assignments to create.')
 param roleAssignments roleAssignmentType[]?
 
+@description('External resource role assignment configuration.')
+type resourceRoleAssignmentType = {
+  @description('Required. The resource ID of the target resource to assign permissions to.')
+  resourceId: string
+  
+  @description('Required. The role definition ID (GUID) or full role definition resource ID. Role names are not supported. Example: "ba92f5b4-2d11-453d-a403-e96b0029c9fe" or "/subscriptions/{sub}/providers/Microsoft.Authorization/roleDefinitions/ba92f5b4-2d11-453d-a403-e96b0029c9fe".')
+  roleDefinitionIdOrName: string
```

Comment:
Should be renamed as the `name` is actually not supported (as per your description). So I guess it's only `roleDefinitionId`